dbms/programs/odbc-bridge/CMakeLists.txt

```diff
@@ -30,6 +30,11 @@ if (Poco_Data_FOUND)
     set(CLICKHOUSE_ODBC_BRIDGE_LINK ${CLICKHOUSE_ODBC_BRIDGE_LINK} PRIVATE ${Poco_Data_LIBRARY})
     set(CLICKHOUSE_ODBC_BRIDGE_INCLUDE ${CLICKHOUSE_ODBC_BRIDGE_INCLUDE} SYSTEM PRIVATE ${Poco_Data_INCLUDE_DIR})
 endif ()
+if (USE_JEMALLOC)
+    # We need to link jemalloc directly to odbc-bridge-library, because in other case
+    # we will build it with default malloc.
+    set(CLICKHOUSE_ODBC_BRIDGE_LINK ${CLICKHOUSE_ODBC_BRIDGE_LINK} PRIVATE ${JEMALLOC_LIBRARIES})
```

Comment:
But we already have executables linked with jemalloc if it's enabled

https://github.com/yandex/ClickHouse/blob/master/CMakeLists.txt#L393

Reply:
Unfortunately `clickhouse-odbc-bridge-lib` goes after this in linker command:
```
/usr/bin/ccache
/usr/bin/g++-9

-fdiagnostics-color=always
-fsized-deallocation

-pipe
-msse4.1
-msse4.2
-mpopcnt

-fno-omit-frame-pointer

-Wall

-Wnon-virtual-dtor
-Wno-array-bounds
-Wextra
-Werror
-Wshadow
-O2
-g
-DNDEBUG
-O3

-fuse-ld=gold
-rdynamic
-Wl,--no-undefined
-Wl,--no-export-dynamic


-rdynamic
dbms/programs/odbc-bridge/CMakeFiles/clickhouse-odbc-bridge.dir/odbc-bridge.cpp.o

-o
dbms/programs/clickhouse-odbc-bridge

dbms/libclickhouse_new_delete.a
contrib/jemalloc-cmake/libjemalloc.a -- problem is here.
dbms/programs/odbc-bridge/libclickhouse-odbc-bridge-lib.a -- aligned_alloc required here.
libs/libdaemon/libdaemon.a
libs/libloggers/loggers/libloggers.a
dbms/libdbms.a
dbms/src/Parsers/libclickhouse_parsers.a
contrib/cppkafka-cmake/libcppkafka.a
contrib/librdkafka-cmake/librdkafka.a
dbms/src/Common/Config/libclickhouse_common_config.a
dbms/src/Common/ZooKeeper/libclickhouse_common_zookeeper.a
dbms/src/Dictionaries/Embedded/libclickhouse_dictionaries_embedded.a
dbms/libclickhouse_common_io.a
dbms/src/Common/StringUtils/libstring_utils.a
libs/libwidechar_width/libwidechar_width.a
contrib/poco/Net/libPocoNet.a
contrib/poco/XML/libPocoXML.a
contrib/re2/libre2.a
contrib/re2_st/libre2_st.a
contrib/boost-cmake/libboost_program_options_internal.a
libs/libcommon/libapple_rt.a
contrib/croaring/libroaring.a
contrib/libcpuid/libcpuid.a
contrib/poco/NetSSL_OpenSSL/libPocoNetSSL.a
contrib/poco/Crypto/libPocoCrypto.a
contrib/poco/Util/libPocoUtil.a
contrib/poco/Crypto/libPocoCrypto.a
contrib/libhdfs3-cmake/libhdfs3.a
contrib/libgsasl/liblibgsasl.a
contrib/libxml2-cmake/liblibxml2.a
contrib/brotli-cmake/libbrotli.a
libs/libmysqlxx/libmysqlxx.a
libs/libcommon/libcommon.a
contrib/cityhash102/libcityhash.a
contrib/cctz-cmake/libcctz.a
libs/libmemcpy/libmemcpy.a
contrib/poco/Util/libPocoUtil.a
contrib/poco/XML/libPocoXML.a
contrib/poco/JSON/libPocoJSON.a
contrib/mariadb-connector-c/libmariadb/libmariadbclient.a
contrib/ssl/ssl/libssl.a
contrib/ssl/crypto/libcrypto.a
contrib/libbtrie/libbtrie.a
contrib/boost-cmake/libboost_filesystem_internal.a
contrib/boost-cmake/libboost_system_internal.a
contrib/poco/MongoDB/libPocoMongoDB.a
contrib/poco/Redis/libPocoRedis.a
contrib/poco/Net/libPocoNet.a
/usr/lib/x86_64-linux-gnu/libicui18n.a
/usr/lib/x86_64-linux-gnu/libicuuc.a
/usr/lib/x86_64-linux-gnu/libicudata.a
contrib/capnproto-cmake/libcapnpc.a
contrib/capnproto-cmake/libcapnp.a
contrib/capnproto-cmake/libkj.a
contrib/arrow-cmake/libparquet_static.a
contrib/arrow-cmake/libarrow_static.a
contrib/double-conversion-cmake/libdouble-conversion.a
contrib/lz4-cmake/liblz4.a
contrib/zstd-cmake/libzstd.a
contrib/snappy/libsnappy.a
contrib/arrow-cmake/libthrift_static.a
contrib/boost-cmake/libboost_regex_internal.a
contrib/protobuf/cmake/libprotobuf.a
contrib/poco/Foundation/libPocoFoundation.a
contrib/poco/Data/ODBC/libPocoDataODBC.a
contrib/poco/Data/libPocoData.a
contrib/unixodbc-cmake/libunixodbc.a
contrib/unixodbc-cmake/libltdl.a
contrib/poco/Data/libPocoData.a
contrib/poco/Foundation/libPocoFoundation.a
contrib/zlib-ng/libz.a
-Wl,--start-group
libs/libglibc-compatibility/libglibc-compatibility.a
-l:libstdc++.a
-l:libstdc++fs.a
contrib/libunwind-cmake/libunwind.a
-Wl,--end-group
-lpthread
-ldl
-lrt
-pthread
-nodefaultlibs
-lgcc

-lc
-lm
-lrt
-lpthread
-ldl
```

Reply:
I wish we can always use `lld`. It treats all `.a` in one group so we don't need to care about link order.

Regarding to this issue ( because libjemalloc.a is really not a dependency but an interposition, cmake cannot handle it nicely), perhaps we can add `aligned_alloc` or other symbols-to-be-interposed to  `dbms/libclickhouse_new_delete.a`, so that ld and gold will use it

Reply:
It will be nice if you implement this! But I think we can merge this PR as hotfix.

Reply:
I tried linking `clickhouse-odbc-bridge` with the exact command you posted, however I didn't see any `aligned_alloc` symbol in the outcome binary. Could you also share the cmake commandline you used?

Reply:
```cmake .. -DLLVM_FOUND=0 -DCMAKE_CXX_COMPILER=/usr/bin/g++-9 -DCMAKE_C_COMPILER=/usr/bin/gcc-9```